src/org/zaproxy/zap/view/widgets/WritableFileChooser.java

```diff
@@ -24,6 +24,11 @@ public WritableFileChooser(File currentDirectory) {
 	@Override
 	public void approveSelection() {
 		File selectedFile = getSelectedFile();
+		
+		if (selectedFile.length() > selectedFile.getUsableSpace()) {
```

Comment:
We could have both, warn about low disk space available (or if it's full) when choosing the file and warn when there's an exception during the generation of the report, the latter definitely. Currently ReportGenerator just catches the IOExceptions and logs them, which is good, but other than the log messages there's no indication that the report generation failed:
```
660393 [AWT-EventQueue-0] ERROR org.parosproxy.paros.extension.report.ReportGenerator  - com.sun.org.apache.xalan.internal.xsltc.TransletException: org.xml.sax.SAXException: No space left on device
java.io.IOException: No space left on device
javax.xml.transform.TransformerException: com.sun.org.apache.xalan.internal.xsltc.TransletException: org.xml.sax.SAXException: No space left on device
java.io.IOException: No space left on device
	at com.sun.org.apache.xalan.internal.xsltc.trax.TransformerImpl.transform(TransformerImpl.java:752)
	at com.sun.org.apache.xalan.internal.xsltc.trax.TransformerImpl.transform(TransformerImpl.java:359)
	at org.parosproxy.paros.extension.report.ReportGenerator.stringToHtml(ReportGenerator.java:118)
	at org.parosproxy.paros.extension.report.ReportLastScan.generate(ReportLastScan.java:74)
	at org.parosproxy.paros.extension.report.ReportLastScan.generateReport(ReportLastScan.java:190)
	at org.parosproxy.paros.extension.report.ExtensionReport$1.actionPerformed(ExtensionReport.java:98)
```

```
781904 [AWT-EventQueue-0] ERROR org.parosproxy.paros.extension.report.ReportGenerator  - No space left on device
java.io.IOException: No space left on device
	at java.io.FileOutputStream.writeBytes(Native Method)
	at java.io.FileOutputStream.write(FileOutputStream.java:345)
	at sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)
	at sun.nio.cs.StreamEncoder.implWrite(StreamEncoder.java:282)
	at sun.nio.cs.StreamEncoder.write(StreamEncoder.java:125)
	at java.io.OutputStreamWriter.write(OutputStreamWriter.java:207)
	at java.io.BufferedWriter.flushBuffer(BufferedWriter.java:129)
	at java.io.BufferedWriter.write(BufferedWriter.java:230)
	at java.io.Writer.write(Writer.java:157)
	at org.parosproxy.paros.extension.report.ReportGenerator.stringToHtml(ReportGenerator.java:181)
	at org.parosproxy.paros.extension.report.ReportLastScan.generate(ReportLastScan.java:74)
	at org.parosproxy.paros.extension.report.ReportLastScan.generateReport(ReportLastScan.java:190)
	at org.parosproxy.paros.extension.report.ExtensionReport$2.actionPerformed(ExtensionReport.java:116)
```